test/quadrature/test_simps.fypp

```diff
@@ -25,135 +19,70 @@ contains
         type(unittest_type), allocatable, intent(out) :: testsuite(:)
 
         testsuite = [ &
-            new_unittest("simps_sp", test_simps_sp), &
-            new_unittest("simps_dp", test_simps_dp), &
-            new_unittest("simps_qp", test_simps_qp), &
-            new_unittest("simps_weights_sp", test_simps_weights_sp), &
-            new_unittest("simps_weights_dp", test_simps_weights_dp), &
-            new_unittest("simps_weights_qp", test_simps_weights_qp), &
-            new_unittest("simps_zero_sp", test_simps_zero_sp), &
-            new_unittest("simps_zero_dp", test_simps_zero_dp), &
-            new_unittest("simps_zero_qp", test_simps_zero_qp), &
-            new_unittest("simps_even_sp", test_simps_even_sp), &
-            new_unittest("simps_even_dp", test_simps_even_dp), &
-            new_unittest("simps_even_qp", test_simps_even_qp), &
-            new_unittest("simps_weights_even_sp", test_simps_weights_even_sp), &
-            new_unittest("simps_weights_even_dp", test_simps_weights_even_dp), &
-            new_unittest("simps_weights_even_qp", test_simps_weights_even_qp), &
-            new_unittest("simps_six_sp", test_simps_six_sp), &
-            new_unittest("simps_six_dp", test_simps_six_dp), &
-            new_unittest("simps_six_qp", test_simps_six_qp) &
+            new_unittest("simps_sp", test_simps_sp) &
+            #:for k1, t1 in REAL_KINDS_TYPES[1:]
```

Comment:
Why does `REAL_KINDS_TYPES` start at 1, instead of at 0? Is it safe (assume that the order changes for whatever reasons, some tests might be ignored)?
My suggestion would be to use the whole vector `REAL_KINDS_TYPES`, even if some tests are duplicated.